api/envoy/config/filter/network/redis_proxy/v2/redis_proxy.proto

```diff
@@ -141,4 +141,23 @@ message RedisProxy {
   // <arch_overview_redis_configuration>` of the architecture overview for recommendations on
   // configuring the backing clusters.
   PrefixRoutes prefix_routes = 5 [(gogoproto.nullable) = false];
+
+  // Authenticate Redis client connections locally by forcing downstream clients to issue a 'Redis
+  // AUTH command <https://redis.io/commands/auth>`_ with this password before enabling any other
+  // command. If an AUTH command's password matches this password, an "OK" response will be returned
+  // to the client. If the AUTH command password does not match this password, then an "ERR invalid
+  // password" error will be returned. If any other command is received before AUTH when this
+  // password is set, then a "NOAUTH Authentication required." error response will be sent to the
+  // client. If an AUTH command is received when the password is not set, then an "ERR Client sent
+  // AUTH, but no password is set" error will be returned.
+  string downstream_auth_password = 6;
```

Comment:
I'll look into the core.DataSource -- as far as your opening questions are concerned, while the auth command is not very rigorous by itself, it is part of a defense in depth strategy for deploying Redis (TLS encryption is another part of that strategy). Not supporting auth will simply prevent folks from deploying Envoy ... at least at Workday (and I imagine many other firms with sensitive information). In this implementation passwords can be updated incrementally, but we would need to use the hot restart capability to rotate passwords. It would be nice to have these passwords as part of the Secret Discovery Service, but that's a whole another level of infrastructure change. Perhaps the core.DataSource is a good way to straddle the static and fully dynamic worlds?

Reply:
I've changed the passwords to be core.DataSource's instead of strings.